webapp/src/components/chat/ChatResourceList.tsx

```diff
@@ -56,24 +79,117 @@ interface TableItem {
+                    sharedBy: "N/A",
+                    createdOn: 0,
+                    tokens: 0,
+                } as ChatMemorySource;
+            })
+            : [];
+
+        setDocumentImporting(importingResources.length > 0);
+        setResources(importingResources);
+
+        void chat.getChatMemorySources(selectedId).then((sources) => {
+            setResources([...importingResources, ...sources]);
         });
+
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [chatId]);
+    }, [selectedId, importingDocuments]);
+
+    const handleImport = () => {
+        const files = documentFileRef.current?.files;
+
+        if (files && files.length > 0) {
+            // Deep copy the FileList into an array so that the function
+            // maintains a list of files to import before the import is complete.
+            const filesArray = Array.from(files);
+
+            setDocumentImporting(true);
+            dispatch(setImportingDocumentsToConversation({
+                importingDocuments: filesArray.map((file) => file.name),
+                chatId: selectedId
+            }));
+
+            chat.importDocument(selectedId, filesArray).finally(() => {
+                setDocumentImporting(false);
+                dispatch(setImportingDocumentsToConversation({
+                    importingDocuments: [],
+                    chatId: selectedId
+                }));
+            });
+        }
+
+        // Reset the file input so that the onChange event will
+        // be triggered even if the same file is selected again.
+        if (documentFileRef.current?.value) {
+            documentFileRef.current.value = '';
+        }
+    };
 
     const { columns, rows } = useTable(resources);
     return (
         <div className={classes.root}>
-            <Table aria-label="External resource table" className={classes.table}>
+            <h2>Documents</h2>
+            <div className={classes.functional}>
+                {/* Hidden input for file upload. Only accept .txt and .pdf files for now. */}
+                <input
+                    type="file"
+                    ref={documentFileRef}
+                    style={{ display: 'none' }}
+                    accept=".txt,.pdf,.md,.jpg,.jpeg,.png,.tif,.tiff"
+                    multiple={true}
+                    onChange={() => {
+                        handleImport();
+                    }}
+                />
+                <Tooltip content="Embed file into chat session" relationship="label">
+                    <Button
+                        className={classes.uploadButton}
+                        icon={<DocumentArrowUp20Regular />}
+                        disabled={documentImporting}
+                        onClick={() => documentFileRef.current?.click()}
+                    >
+                        Upload
+                    </Button>
+                </Tooltip>
+                {documentImporting && <Spinner size="tiny" />}
```

Comment:
since `importingDocuments` an `documentImporting` are always set together, it seems like we could just use `importDocuments` in place of `documentImporting`?

```suggestion
                {importDocuments.length && <Spinner size="tiny" />}
```